### PR TITLE
Add support for controlling HerePro LEDs using lua scripts

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

**Review by IamPete1 — DISMISSED** [review]
Scripting additions look good.  But I suspect we need to add `depends !defined(HAL_BUILD_AP_PERIPH)` to the majority of the existing methods. Because I guess they won't work on periph. Should save a bunch of flash too.

I like the new `handle_rgb_id` would nice to add support for that to the serial LED notify back-ends in the future too.

File: libraries/AP_HAL_ChibiOS/hwdef/HerePro/hwdef.dat [hunk 1]
```diff
@@ -219,3 +221,5 @@ define AP_PERIPH_HAVE_LED TRUE
 define SCRIPTING_HEAP_SIZE (64*1024)
 
 define GPS_MOVING_BASELINE 1
+
+ROMFS scripts/hereproled.lua libraries/AP_Scripting/examples/hereproled.lua
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/HerePro/hwdef.dat:225]: better to put in scripts folder under hwdef folder for board

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp [hunk 2]
```diff
@@ -616,6 +623,95 @@ void AP_UAVCAN::set_buzzer_tone(float frequency, float duration_s)
     _buzzer.pending_mask = 0xFF;
 }
 
+// notify state send
+void AP_UAVCAN::notify_state_send()
```
Comment (tridge) [inline libraries/AP_UAVCAN/AP_UAVCAN.cpp:627]: need a state rate parameter? or use one of the new option bits

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp [hunk 3]
```diff
@@ -616,6 +623,95 @@ void AP_UAVCAN::set_buzzer_tone(float frequency, float duration_s)
...
+        msg.vehicle_state |= 1 << ardupilot::indication::NotifyState::VEHICLE_STATE_VIDEO_RECORDING;
+    }
+    if (AP_Notify::flags.waiting_for_throw) {
+        msg.vehicle_state |= 1 << ardupilot::indication::NotifyState::VEHICLE_STATE_THROW_READY;
+    }
+
+    msg.aux_data_type = ardupilot::indication::NotifyState::VEHICLE_YAW_EARTH_CENTIDEGREES;
+    uint16_t yaw_cd = (uint16_t)(360.0f - degrees(AP::ahrs().get_yaw()))*100.0f;
+    
+    hal.console->printf("NotifyState: yaw_cd=%d, float_yaw=%f\n", yaw_cd, AP::ahrs().get_yaw());
```
Comment (tridge) [inline libraries/AP_UAVCAN/AP_UAVCAN.cpp:707]: debug print should be removed?

File: libraries/AP_UAVCAN/AP_UAVCAN.cpp [hunk 4]
```diff
@@ -616,6 +623,95 @@ void AP_UAVCAN::set_buzzer_tone(float frequency, float duration_s)
...
+    for (uint8_t i=0; i<2; i++) {
```
Comment (tridge) [inline libraries/AP_UAVCAN/AP_UAVCAN.cpp:708]: better having a const uint8_t *data = (uint8_t *)&yaw_cd;

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
need to check change in fw size for f1 and f3 firmwares

**Comment by bugobliterator** [comment]
F103-GPS and F303-GPS has no flash size change whatsoever.

**Comment by bugobliterator** [comment]
> Scripting additions look good. But I suspect we need to add `depends !defined(HAL_BUILD_AP_PERIPH)` to the majority of the existing methods. Because I guess they won't work on periph. Should save a bunch of flash too.
> 
> I like the new `handle_rgb_id` would nice to add support for that to the serial LED notify back-ends in the future too.

Would like to this, but as a separate PR. It would need to be based on separate PERIPH ENABLE flags as well.

**Review by tridge — APPROVED** [review]

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**